File: client/src/models/Event.ts
import { League } from './League';
import { Sport } from './Sport';

export class Event {
  public id: number;
  public name: string;
  public date: string;
  public homeTeamScore: number;
  public awayTeamScore: number;
  public homeTeamOdds: number;
  public awayTeamOdds: number;
  public drawOdds: number;
  public league: League;
  public sport: Sport;
}

File: server/TradeX.DataAccess/Entities/Event.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TradeX.DataAccess.Entities
{
    public class Event
    {
        public int Id { get; set; }

        [Required]
        public string Name { get; set; }

        public DateTimeOffset Date { get; set; }

        public int HomeTeamScore { get; set; }

        public int AwayTeamScore { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal HomeTeamOdds { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal AwayTeamOdds { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal DrawOdds { get; set; }

        public int LeagueId { get; set; }
        public League League { get; set; }
    }
}
